### PR TITLE
Develop pylint

### PR DESCRIPTION
**Pull request by rallytime**
I might have missed some...let's see how this goes.

### CODE REVIEW

File: salt/client/mixins.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def wrapper(*args, **kwargs):
                    }
             pub_data = {}
             # pull out pub_data if you have it
-            for k, v in kwargs.items():
+            for k, v in list(kwargs.items()):
```
Comment (s0undt3ch) [inline salt/client/mixins.py]: Six.iteritems() :smile:

**Comment by rallytime** [comment]
@s0undt3ch  Ok, the changes here are really close, but I am not sure why these two files are failing suddenly now when the associated lines haven't changed since December;

https://jenkins.saltstack.com/job/salt-pr-lint-n/1540/violations/file/salt/scripts.py/
https://jenkins.saltstack.com/job/salt-pr-lint-n/1540/violations/file/salt/cli/caller.py/

Another one I am not sure why it is throwing an error all of a sudden is this one: https://jenkins.saltstack.com/job/salt-pr-lint-n/1540/violations/file/tests/unit/modules/zcbuildout_test.py/

I am also still seeing errors in http.py. I'm not sure what the correct fix should be there. 

The rest of the failures are related to missing __init__.py methods and can be ignored.

**Comment by rallytime** [comment]
Ah, actually the errors in scripts.py and caller.py are not showing up on the branch pylint tests. Maybe that is just a pylint versions incompatibility issue that should be addressed when the __init__ method issue gets cleared up. Same thing with zc_buildout.py. So, it's just the http.py errors that are causing a remaining issue here.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@s0undt3ch nice! Any thoughts on that last pylint error?

**Comment by s0undt3ch** [comment]
Yep:

```python
salt.ext.six.moves.http_client  # pylint: disable=import-error,no-name-in-module
```

**Pull request merged**